Pump 1.17.0 RNs.mdx (#2706)
TykTechnologies/tyk-docs: developer-support/release-notes/pump.mdx

For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v1.17.0).

#### Breaking Changes

No Breaking Changes in this release.

#### Dependencies

##### 3rd Party Dependencies & Tools

| Third Party Dependency                                    | Tested Versions   | Compatible Versions      |
| :-------------------------------------------------------- | :---------------- | :----------------------- | 
| [MongoDB](https://www.mongodb.com/try/download/community) | 5.x, 6.x, and 7.0 | 4.4.x, 5.x, 6.x, and 7.0 |
| [PostgreSQL](https://www.postgresql.org/download/)        | 13.x - 17.x       | 13.x - 17.x              | 
| [Redis](https://redis.io/download/)                       | 6.x - 7.0, 8.x (up to 8.4) | 6.x - 7.x, 8.x (up to 8.4) | 
| [Valkey](https://valkey.io/download/)                     | 8.0.x, 8.1.x      | 7.2.x, 8.0.x, 8.1.x      |                            

Given the time difference between your upgrade and the release of this version, we recommend customers verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations

> Legacy `PMP_MONGO_*` environment variables are now deprecated and do not support Secrets Management references; migrate to the `TYK_PMP_*` equivalents below.

Reviewer: This table only lists `PMP_MONGO_*`. The actual deprecation in tyk-pump#1060 also covers `PMP_MONGOAGG_*` and `PMP_MONGOSEL_*`, which map to `TYK_PMP_PUMPS_MONGOAGGREGATE_META_*` and `TYK_PMP_PUMPS_MONGOSELECTIVE_META_*` respectively. Worth adding those two so Aggregate/Selective pump users don't miss the migration. Neither has an uptime pump equivalent, so a one-line note below the table should cover it without repeating all the rows again.

</Accordion>

</AccordionGroup>

##### Added

<AccordionGroup>
<Accordion title="Secrets Management">
Tyk Pump now supports Secrets Management, letting you reference values from external secrets stores: the Tyk config file, environment variables, local files (including Kubernetes-mounted secrets), Consul, Vault, AWS Secrets Manager, Azure Key Vault, and GCP Secret Manager, in place of plain-text values in Pump's configuration, using a consistent reference syntax:

- `kv://<store>/<path>[#fragment]` for whole-value references.
- `$kv{<store>:<path>[#fragment]}` for embedding a reference inside a larger string.

The `kv` configuration block can itself contain references, for example pulling a Vault token from an environment variable.

References are resolved once, at startup. Individual secrets stores can be marked as `required` and, if they cannot be reached or initialized, Pump will fail to start with a clear error rather than starting with that store silently unavailable.

See [Secrets Management](/docs/tyk-configuration-reference/kv-store) for full details.
</Accordion>

> </AccordionGroup>

Reviewer 2: ```suggestion

<Accordion title='Added GCP IAM authentication for Redis/Valkey'>
Tyk Pump can now authenticate to Google Cloud Memorystore for Redis/Valkey using GCP IAM-based authentication, as an alternative to a static Redis password.

Pump obtains a Google OAuth2 access token from the workload's identity, using Application Default Credentials (ADC) and authenticates with the Memorystore using this token.

This is supported for use with Memorystore for Valkey and Memorystore for Redis Cluster. The legacy non-cluster Memorystore for Redis has no IAM support from Google and continues to use a static password.

The IAM configuration is set in the new `iam_auth` section of Pump's storage config.
</Accordion>
</AccordionGroup>
```

  - ```bash
    docker pull tykio/tyk-pump-docker-pub:v1.17.0
    ```
- Source code tarball for OSS - [GH Tyk Pump Repo](https://github.com/TykTechnologies/tyk-pump/releases/tag/v1.17.0)

#### Changelog
<a id="Changelog-v1.17.0" data-scroll-offset></a>

##### Changed

<AccordionGroup>

<Accordion title='Updated Golang version to 1.26'>
{/* TT-17868 */}
Tyk Pump has been updated to build with Go 1.26, keeping the toolchain current with upstream Go security and performance updates.
</Accordion>

</AccordionGroup>

##### Added

<AccordionGroup>
<Accordion title="Secrets Management">
Tyk Pump now supports Secrets Management, letting you reference values from external secrets stores: the Tyk config file, environment variables, local files (including Kubernetes-mounted secrets), Consul, Vault, AWS Secrets Manager, Azure Key Vault, and GCP Secret Manager, in place of plain-text values in Pump's configuration, using a consistent reference syntax:

- `kv://<store>/<path>[#fragment]` for whole-value references.
- `$kv{<store>:<path>[#fragment]}` for embedding a reference inside a larger string.

> The `kv` configuration block can itself contain references, for example pulling a Vault token from an environment variable.

Reviewer 2: ```suggestion
The secret management configuration block (`kv`) can itself contain references, for example pulling a Vault token from an environment variable.
```

> The secret management configuration block (`kv`) can itself contain references, for example pulling a Vault token from an environment variable.

Reviewer 2: ```suggestion
The secrets management configuration block (`kv`) can itself contain references, for example pulling a Vault token from an environment variable.
```

---

## 1.17 Release Notes

### 1.17.0 Release Notes

#### Release Date xx August 2026

#### Release Highlights

> This release adds support for referencing secrets and configuration values in the Pump configuration file from an external secret store, using the same unified `kv://` / `$kv{}` syntax introduced across other Tyk components this wave — including the ability to reference secrets from within a pump's own `meta` block (for example, a backend's connection string or credentials). As part of this change, the legacy `PMP_MONGO*` environment variable format is now deprecated. This release also extends GCP IAM-based authentication for Redis/Valkey to Pump's storage connection, updates the Go toolchain to Go 1.26, and fixes two issues affecting the Kinesis pump. This release also addresses reported CVEs in dependent libraries.

Reviewer 2: ```suggestion
This release adds [Secrets Management](/tyk-configuration-reference/kv-store) to the Tyk Pump configuration file for the first time, letting you reference values from external secrets stores in place of plain-text values.

Tyk Pump now supports IAM-based authentication for Google Cloud Memorystore for Redis Cluster and Valkey, as an alternative to a static password.

We have also fixed two issues affecting the Kinesis pump and deprecated the legacy `PMP_MONGO*` environment variable format in lieu of the standard `TYK_PMP_*` format.
```

For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v1.17.0).

#### Breaking Changes

No Breaking Changes in this release.

#### Dependencies

##### 3rd Party Dependencies & Tools

| Third Party Dependency                                    | Tested Versions   | Compatible Versions      |
| :-------------------------------------------------------- | :---------------- | :----------------------- | 
| [MongoDB](https://www.mongodb.com/try/download/community) | 5.x, 6.x, and 7.0 | 4.4.x, 5.x, 6.x, and 7.0 |
| [PostgreSQL](https://www.postgresql.org/download/)        | 13.x - 17.x       | 13.x - 17.x              | 
| [Redis](https://redis.io/download/)                       | 6.x - 7.0, 8.x (up to 8.4) | 6.x - 7.x, 8.x (up to 8.4) | 
| [Valkey](https://valkey.io/download/)                     | 8.0.x, 8.1.x      | 7.2.x, 8.0.x, 8.1.x      |                            

Given the time difference between your upgrade and the release of this version, we recommend customers verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations

> There are no deprecations in this release.

Reviewer 2: ```suggestion
Legacy `PMP_MONGO_*` environment variables are now deprecated and do not support Secrets Management references; migrate to the `TYK_PMP_*` equivalents below.

| Legacy (deprecated) | Use for [standard Mongo pump](/api-management/dashboard-analytics/control-plane-pumps#standard-mongo-pump) | Use for [Mongo uptime pump](/planning-for-production/ensure-high-availability/uptime-tests#mongodb) |
|---|---|---|
| `PMP_MONGO_MONGOURL` | `TYK_PMP_PUMPS_MONGO_META_MONGOURL` | `TYK_PMP_UPTIMEPUMPCONFIG_MONGOURL` |
| `PMP_MONGO_MONGOUSESSL` | `TYK_PMP_PUMPS_MONGO_META_MONGOUSESSL` | `TYK_PMP_UPTIMEPUMPCONFIG_MONGOUSESSL` |
| `PMP_MONGO_MONGOSSLINSECURESKIPVERIFY` | `TYK_PMP_PUMPS_MONGO_META_MONGOSSLINSECURESKIPVERIFY` | `TYK_PMP_UPTIMEPUMPCONFIG_MONGOSSLINSECURESKIPVERIFY` |
| `PMP_MONGO_MONGOSSLALLOWINVALIDHOSTNAMES` | `TYK_PMP_PUMPS_MONGO_META_MONGOSSLALLOWINVALIDHOSTNAMES` | `TYK_PMP_UPTIMEPUMPCONFIG_MONGOSSLALLOWINVALIDHOSTNAMES` |
| `PMP_MONGO_MONGOSSLCAFILE` | `TYK_PMP_PUMPS_MONGO_META_MONGOSSLCAFILE` | `TYK_PMP_UPTIMEPUMPCONFIG_MONGOSSLCAFILE` |
| `PMP_MONGO_MONGOSSLPEMKEYFILE` | `TYK_PMP_PUMPS_MONGO_META_MONGOSSLPEMKEYFILE` | `TYK_PMP_UPTIMEPUMPCONFIG_MONGOSSLPEMKEYFILE` |
| `PMP_MONGO_OMITINDEXCREATION` | `TYK_PMP_PUMPS_MONGO_META_OMITINDEXCREATION` | `TYK_PMP_UPTIMEPUMPCONFIG_OMITINDEXCREATION` |
| `PMP_MONGO_MONGOSESSIONCONSISTENCY` | `TYK_PMP_PUMPS_MONGO_META_MONGOSESSIONCONSISTENCY` | `TYK_PMP_UPTIMEPUMPCONFIG_MONGOSESSIONCONSISTENCY` |
| `PMP_MONGO_MONGODRIVERTYPE` | `TYK_PMP_PUMPS_MONGO_META_MONGODRIVERTYPE` | `TYK_PMP_UPTIMEPUMPCONFIG_MONGODRIVERTYPE` |
| `PMP_MONGO_MONGODIRECTCONNECTION` | `TYK_PMP_PUMPS_MONGO_META_MONGODIRECTCONNECTION` | `TYK_PMP_UPTIMEPUMPCONFIG_MONGODIRECTCONNECTION` |
| `PMP_MONGO_COLLECTIONNAME` | `TYK_PMP_PUMPS_MONGO_META_COLLECTIONNAME` | `TYK_PMP_UPTIMEPUMPCONFIG_COLLECTIONNAME` |
| `PMP_MONGO_MAXINSERTBATCHSIZEBYTES` | `TYK_PMP_PUMPS_MONGO_META_MAXINSERTBATCHSIZEBYTES` | `TYK_PMP_UPTIMEPUMPCONFIG_MAXINSERTBATCHSIZEBYTES` |
| `PMP_MONGO_MAXDOCUMENTSIZEBYTES` | `TYK_PMP_PUMPS_MONGO_META_MAXDOCUMENTSIZEBYTES` | `TYK_PMP_UPTIMEPUMPCONFIG_MAXDOCUMENTSIZEBYTES` |
| `PMP_MONGO_COLLECTIONCAPMAXSIZEBYTES` | `TYK_PMP_PUMPS_MONGO_META_COLLECTIONCAPMAXSIZEBYTES` | `TYK_PMP_UPTIMEPUMPCONFIG_COLLECTIONCAPMAXSIZEBYTES` |
| `PMP_MONGO_COLLECTIONCAPENABLE` | `TYK_PMP_PUMPS_MONGO_META_COLLECTIONCAPENABLE` | `TYK_PMP_UPTIMEPUMPCONFIG_COLLECTIONCAPENABLE` |

```

#### Upgrade instructions

For users currently on v1.16.0, we strongly recommend promptly upgrading to the latest release.

Go to the [Upgrading Tyk](#upgrading-tyk) section for detailed upgrade Instructions.

#### Downloads
- [Docker Image v1.17.0](https://hub.docker.com/r/tykio/tyk-pump-docker-pub/tags?page=&page_size=&ordering=&name=v1.17.0)
  - ```bash
    docker pull tykio/tyk-pump-docker-pub:v1.17.0
    ```
- Source code tarball for OSS - [GH Tyk Pump Repo](https://github.com/TykTechnologies/tyk-pump/releases/tag/v1.17.0)

#### Changelog
<a id="Changelog-v1.17.0" data-scroll-offset></a>

##### Changed

<AccordionGroup>

<Accordion title='Updated Golang version to 1.26'>
{/* TT-17868 */}
Tyk Pump has been updated to build with Go 1.26, keeping the toolchain current with upstream Go security and performance updates.
</Accordion>

</AccordionGroup>

> 

Reviewer 2: ```suggestion

##### Added

<AccordionGroup>
<Accordion title="Secrets Management">
Tyk Pump now supports Secrets Management, letting you reference values from external secrets stores: the Tyk config file, environment variables, local files (including Kubernetes-mounted secrets), Consul, Vault, AWS Secrets Manager, Azure Key Vault, and GCP Secret Manager, in place of plain-text values in Pump's configuration, using a consistent reference syntax:

- `kv://<store>/<path>[#fragment]` for whole-value references.
- `$kv{<store>:<path>[#fragment]}` for embedding a reference inside a larger string.

The `kv` configuration block can itself contain references, for example pulling a Vault token from an environment variable.

References are resolved once, at startup. Individual secrets stores can be marked as `required` and, if they cannot be reached or initialized, Pump will fail to start with a clear error rather than starting with that store silently unavailable.

See [Secrets Management](/docs/tyk-configuration-reference/kv-store) for full details.
</Accordion>
</AccordionGroup>

```
